### PR TITLE
Replaced Stringbuilder by String

### LINKED ISSUE #3335: AbstractPortsExtractor: 'StringBuilder' can be replaced with 'String'
### Description

The class [AbstractPortsExtractor](https://github.com/eclipse-jkube/jkube/blob/5da6068d7756784aef9569568ee80f98da97e296/jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractor.java#L156) presents the issue: 'StringBuilder' can be replaced with 'String'.

The following line:
https://github.com/eclipse-jkube/jkube/blob/5da6068d7756784aef9569568ee80f98da97e296/jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractor.java#L156
should be changed to:

```java
					flat.put(stringKey + DOT + entry.getKey(),
```
### :red_circle: Before you start :red_circle:

:point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down:

Make sure you read the [contributing guide](https://www.eclipse.org/jkube/contributing) first.

Pay special attention to the ECA agreement section and the requirement to sign-off your commit.

This is a [first-timers-only](https://eclipse.dev/jkube/contributing/#first-timers-only-issues) issue. These issues are especially created for contributors who are contributing to the project for the **first time**.

If you have already contributed to this project in the past, please don't ask to be assigned one of these issues and look for [good-first-issues](https://github.com/eclipse-jkube/jkube/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22) or [help-wanted](https://github.com/eclipse-jkube/jkube/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22help+wanted%22+) issues instead in order to continue you contributor journey in the project.

### PR DESCRIPTION
**Pull request by HarshitMadan77**
## Description
Replaced 'Stringbuilder' by 'String'
Fixes #3335 

## Type of change
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractor.java [hunk 1]
```diff
@@ -122,7 +122,8 @@ public Map<String, Integer> extract(JavaProject project) {
 	 */
 	private Map<String, String> readConfig(File f) throws IOException {
 		Map<String, String> map;
-		if (f.getName().endsWith(JSON_EXTENSION) || f.getName().endsWith(YAML_EXTENSION) || f.getName().endsWith(YML_EXTENSION)) {
+		if (f.getName().endsWith(JSON_EXTENSION) || f.getName().endsWith(YAML_EXTENSION)
```
Comment (rohanKanojia) [inline jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractor.java:125]: How are these changes related to #3335

Comment (rohanKanojia) [reply jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractor.java:125]: @HarshitMadan77 : Could you please remove changes to this if block?

**Comment by FabioDevGomes** [comment]
can i work on this issue? i already have the project on my windows and a read the contributing guidelines. Not all tests passed but I believe that this should not influence this issue.
